ui/src/components/FormModifications/FormModifications.ts

```diff
@@ -0,0 +1,124 @@
+import { Mode } from '../../constants/modes';
+import { AcceptableFormValueOrNullish } from '../../types/components/shareableTypes';
+import { BaseFormState, AnyEntity, EntitiesWithModifications } from '../BaseFormTypes';
+import { MarkdownMessageProps } from '../MarkdownMessage/MarkdownMessage';
+
+export const handleStateFieldModificationProp = (
+    key: 'display' | 'value' | 'disabled' | 'markdownMessage',
+    propValue: string | number | boolean | object,
+    fieldId: string,
+    state: BaseFormState
+) => {
+    const shallowStateCopy = { ...state };
+    if (shallowStateCopy.data) {
+        shallowStateCopy.data[fieldId][key] = propValue as AcceptableFormValueOrNullish &
+            boolean &
+            (MarkdownMessageProps | undefined);
+        return {
+            changesOccured: true,
+            data: shallowStateCopy,
+        };
+    }
+    return {
+        changesOccured: false,
+        data: shallowStateCopy,
+    };
+};
+
+export const handleEntityModificationProp = (
+    key: 'help' | 'label',
+    propValue: string,
+    fieldId: string,
+    state: BaseFormState
+) => {
+    const shallowStateCopy = { ...state };
+    if (shallowStateCopy?.data?.[fieldId]) {
+        shallowStateCopy.data[fieldId].modifiedEntitiesData = {
+            ...shallowStateCopy.data[fieldId].modifiedEntitiesData,
+            [key]: propValue,
+        };
+        return {
+            changesOccured: true,
+            data: shallowStateCopy,
+        };
+    }
+
+    return {
+        changesOccured: false,
+        data: shallowStateCopy,
+    };
+};
+
+export function getAllFieldsWithModifications(entities: AnyEntity[]): EntitiesWithModifications[] {
+    const entitiesWithModifications = entities.filter(
+        (e) =>
+            e.type !== 'helpLink' &&
+            e.type !== 'checkboxGroup' &&
+            e.type !== 'oauth' &&
+            e.type !== 'custom' &&
+            e?.modifyFieldsOnValue
+    );
+
+    return entitiesWithModifications as EntitiesWithModifications[];
+}
+export const getModifiedState = (
+    state: BaseFormState,
+    mode: Mode,
+    entitiesToModify: EntitiesWithModifications[]
+) => {
+    let stateShallowCopy = { ...state };
+    let shouldUpdateState = false;
+
+    entitiesToModify?.forEach((entity: EntitiesWithModifications) => {
+        let modification = entity?.modifyFieldsOnValue?.find(
```

Comment:
seems like `EntitiesWithModifications` is not strict enough because I see that we expect `modifyFieldsOnValue` always present (since there is a check in `getAllFieldsWithModifications`)

Reply:
We can change it to call `getAllFieldsWithModifications` inside `getModifiedState`, to filter out `entitiesToModify` property at the beginning, but actually it is just needed only on initial load, 
for rest of them(after each field is changed) the doubt in presence of those modifications is needed as we do not filter them out in BaseFormView file

also changed `EntitiesWithModifications` into `EntitiesAllowingModifications`